### PR TITLE
restore ability to hide scripts panel

### PR DESCRIPTION
**Pull request by kubecz3k**
script_panel_any_x_size

### CODE REVIEW

**Comment by kubecz3k** [comment]
It's still impossible to squash VS (which from what reduz said is good I think): 
vs_script_panel_squash

**Comment by reduz** [comment]
this is why i wanted to add a hide button instead of allowing this feature

File: editor/plugins/script_editor_plugin.cpp [hunk 1]
```diff
@@ -2182,7 +2182,7 @@ ScriptEditor::ScriptEditor(EditorNode *p_editor) {
 
 	script_list = memnew(ItemList);
 	list_split->add_child(script_list);
-	script_list->set_custom_minimum_size(Size2(150 * EDSCALE, 100)); //need to give a bit of limit to avoid it from disappearing
+	script_list->set_custom_minimum_size(Size2(0, 100)); //need to give a bit of limit to avoid it from disappearing
```
Comment (vnen) [inline editor/plugins/script_editor_plugin.cpp]: You should remove the comment. It doesn't make sense if the list can disappear.

Comment (kubecz3k) [inline editor/plugins/script_editor_plugin.cpp]: I think it makes sense, identical comment is in line 2192: `members_overview->set_custom_minimum_size(Size2(0, 100))` and there is 0 in X. I think it's about Y size (you can see on gif it's impossible to squash those panels vertically). Am I wrong?

**Comment by kubecz3k** [comment]
@reduz would be cool to have such button

**Comment by kubecz3k** [comment]
also would be cool if this button could remember states independently for gdscript and VS

**Comment by kubecz3k** [comment]
hmm not sure where exactly that that button could go, maybe an option in file menu + keyboard shortcut  would be enough?

**Comment by reduz** [comment]
May be a menu option with shortcut I guess

File: editor/plugins/script_editor_plugin.cpp (hunk 1 above)
Comment (vnen) [inline editor/plugins/script_editor_plugin.cpp]: I guess you're right but the comment is not really clear about that. [When it was added](https://github.com/godotengine/godot/commit/2a3e00c8c7e6f997b12864755a3df3b9bd3cca05#diff-5b09686358173e1d947a01511212a3ebR2182) the previous minimum size was (0, 0).

**Comment by kubecz3k** [comment]
Created new pr to solve that use case: #9588

**Pull request closed without merging**